plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergGlueCatalogAccessOperations.java

```diff
@@ -0,0 +1,456 @@
+import static io.trino.testing.sql.TestTable.randomTableSuffix;
+import static java.lang.String.format;
+import static java.lang.String.join;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.fail;
+
+/*
+ * TestIcebergGlueCatalogAccessOperations currently uses AWS Default Credential Provider Chain,
+ * See https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#credentials-default
+ * on ways to set your AWS credentials which will be needed to run this test.
+ */
+@Test(singleThreaded = true)
+public class TestIcebergGlueCatalogAccessOperations
+        extends AbstractTestQueryFramework
+{
+    private static AtomicReference<GlueMetastoreStats> catalogStatsPin;
+    private static AtomicReference<GlueMetastoreStats> tableOperationStatsPin;
+
+    private final String schema = "test_schema_" + randomTableSuffix();
+    private final Session TEST_SESSION = testSessionBuilder()
+            .setCatalog("iceberg")
+            .setSchema(schema)
+            .build();
+
+    private GlueMetastoreStats catalogStats;
+    private GlueMetastoreStats tableOperationStats;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        File tmp = Files.createTempDirectory("test_iceberg").toFile();
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(TEST_SESSION).build();
+        Plugin plugin = new TestingIcebergPlugin(Optional.empty(), Optional.empty(), Optional.of(new StealStatsModule()));
+
+        queryRunner.installPlugin(plugin);
+        queryRunner.createCatalog("iceberg", "iceberg",
+                ImmutableMap.of(
+                        "iceberg.catalog.type", "glue",
+                        "hive.metastore.glue.default-warehouse-dir", tmp.getAbsolutePath()
+                ));
+        queryRunner.execute("CREATE SCHEMA " + schema);
+        catalogStats = catalogStatsPin.get();
+        tableOperationStats = tableOperationStatsPin.get();
+        return queryRunner;
+    }
+
+    @Test
+    public void testCreateTable()
+    {
+        try {
+            assertMetastoreInvocations("CREATE TABLE test_create (id VARCHAR, age INT)",
+                    ImmutableMultiset.builder()
+                            .add(CREATE_TABLE)
+                            .add(GET_DATABASE)
+                            .add(GET_TABLE)
+                            .build());
+        }
+        finally {
+            getQueryRunner().execute("DROP TABLE IF EXISTS test_create");
+        }
+    }
+
+    @Test
+    public void testCreateTableAsSelect()
+    {
+        try {
+            assertMetastoreInvocations("CREATE TABLE test_ctas AS SELECT 1 AS age",
+                    ImmutableMultiset.builder()
+                            .add(GET_DATABASE)
+                            .add(CREATE_TABLE)
+                            .add(GET_TABLE)
+                            .build());
+        }
+        finally {
```

Comment:
you can drop `try-finally` because there is no risk of table name collisions with other tests.

Reply:
The original error that was causing poor table cleanup was the assertion errors for the tests. I want to be able to clean up the schema even in the cases where the counts are off